### ISSUE TITLE
Exclude certain files from being copied

### ISSUE THREAD

**Issue by oogali** [nickjj/esbuild-copy-static-files#4]
I've been using this plugin for a month now to simplify publishing static files into the resulting build and I'm pleased with it so far. Due to some quirks in the project I'm working on, I need to exclude certain files from being copied.

While I know I can override the filter function as an optional argument, it would require me to re-implement the entire logic of filter function which seems a bit excessive.

I can spin up a PR to add this functionality but I'm looking for guidance for the best way to implement this...

My thoughts are I could take one of the following approaches:

1. Add a new `exclude` argument to the options object and have the built-in `filter` function evaluate that when it's called (e.g. if source file name with leading directory stripped matches an element in the excluded array, then return false).

2. Export the `filter` function to allow it to be reused externally, then one doesn't have to choose between your filter logic and their filter logic.

The problem I see with option 1 is this would make the API and documentation a bit clunky since it requires a caveat in giant bold letters that `exclude` will be ignored if the `filter` function is overridden.

Whether option 2 is palatable is up to you and your preferences.

What are your thoughts?

**Comment by nickjj** [comment]
Hi,

If you were to re-implement the filter function what would it look like?

**Comment by oogali** [comment]
What I'd prefer to do is something along the lines of...

```
import copyStaticFiles, { filter } from 'esbuild-copy-static-files'

const EXCLUDED_FILES = [...]

function excludeFilter(src, dst) {
  if (EXCLUDED_FILES.find(f => src === f) {
    return false;
  }

  return filter(src, dst)
}
```

```
...
  plugins: [
    copyStaticFiles({ filter: excludeFilter })
  ]
}
```

**Comment by oogali** [comment]
Otherwise, to cover my use case, I'd end up duplicating all of your `filter()` code and the functions it depends on into my external function just to pass it back in to your scope as an argument.

**Comment by ianchanning** [comment]
Yes - I think this would be an excellent addition - exporting the `filter` function to be able to re-use it is very useful.

**Comment by ianchanning** [comment]
Ha, I'm actually looking into re-using the `filter` function and I've just realised that 90% of this plugin *is* the `filter` function and its dependencies.

**Comment by nickjj** [comment]
Yep! Feel free to open a PR if you want to arrive at a solution that lets you:

- Define a custom filter function (current behavior)
- Augment / extend the existing filter function with a new filter function of your choosing
